### PR TITLE
feat: [CDS-76624]: Configuration screen new split view

### PR DESCRIPTION
**Pull request by MondalSayantan**
### Summary

 - Added `Read Only` mode for new left nav
 - Minor improvements to config page like setting border-radius to 0px and white colour on hovering over nav modules.

#### Screenshots

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/10-common/navigation/ModuleConfigurationScreen/ModuleDetailsSection/ModuleCarousel.module.scss [hunk 1]
```diff
@@ -66,10 +52,16 @@
...
+  box-sizing: border-box;
+  border: 20px solid var(--primary-1);
+  box-shadow: 0px 0px 0px 1px var(--grey-200) inset;
+  border-radius: 0px;
 }
 
-.iconHeading {
-  top: var(--spacing-huge) !important;
+.readOnlyContainer {
+  border: 20px solid #f6fcff; //Not using variables here because color not available in uicore
```
Comment (Sagar-Rajput7) [inline src/modules/10-common/navigation/ModuleConfigurationScreen/ModuleDetailsSection/ModuleCarousel.module.scss:65]: Hex-codes are not allowed as per linting, let use similar uicore color or rgb.

@avinashmadhwani02  thoughts?

File: src/modules/10-common/navigation/ModuleList/NavModule/NavModule.tsx [hunk 2]
```diff
@@ -44,20 +46,29 @@ const navModuleToClassMap: Record<NavModuleName, string> = {
...
+        className,
         { [css.lightContainer]: theme === 'LIGHT' },
+        { [css.active]: active },
         navModuleToClassMap[module]
       )}
       flex={{ justifyContent: 'space-between' }}
       padding={{ top: 'small', bottom: 'small', left: 'large', right: 'large' }}
       background={Color.PRIMARY_9}
-      width={checkboxProps ? 348 : 310}
+      width={checkboxProps ? 348 : readOnly ? 348 : 278}
```
Comment (Sagar-Rajput7) [inline src/modules/10-common/navigation/ModuleList/NavModule/NavModule.tsx:71]:
```suggestion
      width={checkboxProps || readonly ? 348 : 278}
```

**Review by Sagar-Rajput7 — APPROVED** [review]

File: src/modules/10-common/navigation/ModuleConfigurationScreen/ModuleDetailsSection/ModuleCarousel.module.scss (hunk 1 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/navigation/ModuleConfigurationScreen/ModuleDetailsSection/ModuleCarousel.module.scss:65]: We can ask UX team to provide color from design system

File: src/modules/10-common/navigation/ModuleConfigurationScreen/CarousellmageAndDescription/CarousellmageAndDescription.module.scss [hunk 3]
```diff
@@ -28,3 +28,25 @@
   opacity: 0.14;
   z-index: -1;
 }
+
+.lightContainer {
+  .primaryText {
+    flex-wrap: wrap;
```
Comment (avinashmadhwani02) [inline src/modules/10-common/navigation/ModuleConfigurationScreen/CarousellmageAndDescription/CarousellmageAndDescription.module.scss:34]: why do we need flex-wrap for text?

**Comment by MondalSayantan** [comment]
retrigger jest

**Comment by MondalSayantan** [comment]
retrigger jest

**Comment by MondalSayantan** [comment]
retrigger cypress-pipeline

**Comment by MondalSayantan** [comment]
retrigger cypress-pipeline

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by Sagar-Rajput7 — DISMISSED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/10-common/navigation/ModuleConfigurationScreen/ModuleDetailsSection/__tests__/__snapshots__/ModuleCarousel.test.tsx.snap [hunk 4]
```diff
@@ -7,68 +7,64 @@ exports[`Module Carousel test render without data 1`] = `
...
-      <div
-        class="Layout--horizontal StyledProps--main heading"
+      <span
+        class="bp3-icon StyledProps--main StyledProps--margin-right-xsmall"
+        data-icon="cd-main"
       >
-        <span
-          class="bp3-icon StyledProps--main StyledProps--margin-right-xsmall"
-          data-icon="cd-main"
+        <svg
```
[the remark this answers is not in the document]
Comment (MondalSayantan) [reply src/modules/10-common/navigation/ModuleConfigurationScreen/ModuleDetailsSection/__tests__/__snapshots__/ModuleCarousel.test.tsx.snap:16]: It was a miss from my part in this PR https://github.com/harness/harness-core-ui/pull/17443 . I have reverted those changes so the snapshots are updated again

**Review by Sagar-Rajput7 — APPROVED** [review]

**Pull request merged**